### ISSUE TITLE
Small contigs cause reverse blast hit results and trim fail

### ISSUE THREAD

**Issue by tseemann** [tseemann/berokka#8]
Usually the beginning hits the end, which we handle:
```
Running: blastn -query 4.head.fa -subject 4.fa -out 4.bls -evalue 1E-6 -dust no
blastn: 1..13766/20000 aligns to 43298..57075/57075
tig00000003 keep 1..43297/57075 (remove 13779 bp)
```

On these smaller ones, the end hits the beginning, which we DO NOT HANDLE.
```
*** [7] tig00000006 ***
Using first 900 bp to BLAST
Writing tig00000006 ( 900 bp ) to 7.fa
Writing tig00000006 ( 900 bp ) to 7.head.fa
Running: blastn -query 7.head.fa -subject 7.fa -out 7.bls -evalue 1E-6 -dust no
blastn: 781..900/900 aligns to 1..120/900
tig00000006 - COULD NOT TRIM
```

**Comment by tseemann** [comment]
Turns out it's the 3rd HSP (1st is self-self, 2nd is end2-end1, 3rd is end1-end2).

```
 Score = 222 bits (120),  Expect = 7e-62
 Identities = 120/120 (100%), Gaps = 0/120 (0%)
 Strand=Plus/Plus

Query  781  GGTGCTGGGGCGTCTGTATCGGGAAAACCTGAGCTTTAACGAGGCGGTGCGCGCCGGGCT  840
            ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  1    GGTGCTGGGGCGTCTGTATCGGGAAAACCTGAGCTTTAACGAGGCGGTGCGCGCCGGGCT  60

Query  841  GTTTACCATCCCAGGCGACGGCTGCATCGACTATGCGCCGATCCTCGACTTTGTCCGGGA  900
            ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  61   GTTTACCATCCCAGGCGACGGCTGCATCGACTATGCGCCGATCCTCGACTTTGTCCGGGA  120


 Score = 222 bits (120),  Expect = 7e-62
 Identities = 120/120 (100%), Gaps = 0/120 (0%)
 Strand=Plus/Plus

Query  1    GGTGCTGGGGCGTCTGTATCGGGAAAACCTGAGCTTTAACGAGGCGGTGCGCGCCGGGCT  60
            ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  781  GGTGCTGGGGCGTCTGTATCGGGAAAACCTGAGCTTTAACGAGGCGGTGCGCGCCGGGCT  840

Query  61   GTTTACCATCCCAGGCGACGGCTGCATCGACTATGCGCCGATCCTCGACTTTGTCCGGGA  120
            ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  841  GTTTACCATCCCAGGCGACGGCTGCATCGACTATGCGCCGATCCTCGACTTTGTCCGGGA  900
```

**Comment by tseemann** [comment]
Fixed, will be in 0.2
